nipype/utils/filemanip.py

```diff
@@ -314,38 +320,30 @@ def copyfile(originalfile, newfile, copy=False, create_new=False,
             fmlogger.warn(e.message)
 
     # Associated files
-    if originalfile.endswith(".img"):
-        hdrofile = originalfile[:-4] + ".hdr"
-        hdrnfile = newfile[:-4] + ".hdr"
-        matofile = originalfile[:-4] + ".mat"
-        if os.path.exists(matofile):
-            matnfile = newfile[:-4] + ".mat"
-            copyfile(matofile, matnfile, copy, hashmethod=hashmethod,
-                     use_hardlink=use_hardlink)
-        copyfile(hdrofile, hdrnfile, copy, hashmethod=hashmethod,
-                 use_hardlink=use_hardlink)
-    elif originalfile.endswith(".BRIK"):
-        hdrofile = originalfile[:-5] + ".HEAD"
-        hdrnfile = newfile[:-5] + ".HEAD"
-        copyfile(hdrofile, hdrnfile, copy, hashmethod=hashmethod,
-                 use_hardlink=use_hardlink)
+    _, _, this_type = split_filename(originalfile)
+    for type_set in related_filetype_sets:
+        if this_type in type_set:
+            for alt_type in type_set:
+                alt_ofile = originalfile[:-len(this_type)] + alt_type
+                alt_nfile = newfile[:-len(this_type)] + alt_type
+                if os.path.exists(alt_ofile) and not os.path.exists(alt_nfile):
```

Comment:
if original and new files exists, the function assumes we have already done the copy. This is therefore, the recursive base condition. This was my attempt at minimising changes to the function signature. Probably a better way is to change the signature:

    def copyfile(originalfile, newfile, copy=False, create_new=False,
                 hashmethod=None, use_hardlink=False, copy_related_files=True)

And then, when we recurse for related files, set `copy_related_files` to `False`